hub/api/dataset.py

```diff
@@ -220,19 +220,29 @@ def __iter__(self):
             yield self[i]
 
     def _load_meta(self):
-        if dataset_exists(self.storage):
+        meta_key = get_dataset_meta_key()
+
+        if meta_key in self.storage:
+            # dataset exists
+
             logger.info(f"Hub Dataset {self.path} successfully loaded.")
-            self.meta = DatasetMeta.load(self.storage)
+            self.meta = self.storage.get_cachable(meta_key, DatasetMeta)
             for tensor_name in self.meta.tensors:
                 self.tensors[tensor_name] = Tensor(tensor_name, self.storage)
         elif len(self.storage) > 0:
+            # dataset does not exist, but the path was not empty
+
             raise PathNotEmptyException
         else:
-            self.meta = DatasetMeta.create(self.storage)
+            # dataset does not exist
+
+            self.meta = DatasetMeta()
+            self.storage[meta_key] = self.meta
+
             self.flush()
             if self.path.startswith("hub://"):
                 self.client.create_dataset_entry(
-                    self.org_id, self.ds_name, self.meta.to_dict(), public=self.public
+                    self.org_id, self.ds_name, self.meta.__dict__, public=self.public
```

Comment:
Are we sure we want to include all attributes here? Since it inherits from Cachable, if that ever has state will it be saved here as well? I like the idea of `to_dict` just in case.

Reply:
makes sense